src/Microsoft.AspNet.OData.Shared/NavigationPath.cs

```diff
@@ -0,0 +1,90 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="NavigationPath.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Navigation Path of an OData ID
+    /// </summary>
+    public class NavigationPath
+    {
+        private string _navigationPathName;
+        private ReadOnlyCollection<ODataPathSegment> _pathSegments;
+        PathItem[] _pathItems;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NavigationPath"/> class.
+        /// </summary>        
+        /// <param name="pathSegments">Pathsegment collection</param>
+        public NavigationPath(ReadOnlyCollection<ODataPathSegment> pathSegments)
+        {
+             Debug.Assert(pathSegments.Count > 0);
+
+            _pathSegments = pathSegments;
+            _navigationPathName = pathSegments.First().Identifier;
```

Comment:
>_navigationPathName [](http://example.com/codeflow?start=12&length=19)

Is this always right? 
What does the collection of path segments represent? Will it always start with a top level entityset or singleton (or function import?) where/how is _navigationPathName used?